File: 0140-word-break-ii/0140-word-break-ii.js
/**
 * @param {string} s
 * @param {string[]} wordDict
 * @return {string[]}
 */
var wordBreak = function(s, wordDict) {
    const f = (i) => {
        let result = wordDict.includes(s.slice(i)) ? [s.slice(i)] : [];
        for (let j = i + 1; j < s.length; j++) {
            if (wordDict.includes(s.slice(i, j))) {
                for (let t of f(j)) {
                    result.push(s.slice(i, j) + ' ' + t);
                }
            }
        }
        return result;
    };

    return f(0);
};

File: 0140-word-break-ii/0140-word-break-ii.py
class Solution:
    def wordBreak(self, s: str, wordDict: List[str]) -> bool:
        return (f:=lambda i:[s[i:]]*(s[i:] in wordDict)+[s[i:j]+' '+t for j in range(i+1,len(s)) if s[i:j] in wordDict for t in f(j)])(0)